### PR TITLE
ACP-236 (6): Add reward validator tests for auto-renewed validators

### LINKED ISSUE #4499: Continuous Staking - Implementation

### PR DESCRIPTION
**Pull request by rrazvan1**
## Why this should be merged
Adds test coverage for the reward/renewal flow introduced in `razvan/acp-236-s5`.

## How this works
- Tests reward distribution, auto-compounding shares, validator re-insertion after period expiry, and edge cases.

## Need to be documented in RELEASES.md?
No

### CODE REVIEW

File: vms/platformvm/txs/executor/reward_validator_test.go [hunk 1]
```diff
@@ -887,3 +889,863 @@ func TestRewardDelegatorTxExecuteOnAbort(t *testing.T) {
 	require.NoError(err)
 	require.Equal(initialSupply-expectedReward, newSupply, "should have removed un-rewarded tokens from the potential supply")
 }
+
+func TestRewardValidatorStakerType(t *testing.T) {
```
Comment (joshua-kim) [inline vms/platformvm/txs/executor/reward_validator_test.go:893]: Should we merge this into the proposal tx PR? This is the UT to validate the proposal changes so I think they should be reviewed together.

Comment (rrazvan1) [reply vms/platformvm/txs/executor/reward_validator_test.go:893]: makes sense, will move it.

**Review by DracoLi — APPROVED** [review]

File: vms/platformvm/txs/executor/reward_validator_test.go [hunk 2]
```diff
@@ -944,3 +944,808 @@ func TestRewardValidatorStakerType(t *testing.T) {
 	)
 	require.ErrorIs(t, err, errShouldUseRewardAutoRenewedValidator)
 }
+
+func TestRewardAutoRenewedValidatorTxErrors(t *testing.T) {
+	var (
+		env           = newEnvironment(t, upgradetest.Fortuna)
```
Comment (yacovm) [inline vms/platformvm/txs/executor/reward_validator_test.go]: shouldn't we be using the latest upgrade everywhere?

**Comment by joshua-kim** [comment]
Can we merge this directly into https://github.com/ava-labs/avalanchego/pull/5204? I think that the tests should be in the same PR because they validate the changes introduced, and although it makes the resulting diff bigger I don't think it is actually more difficult to review because the tests are required to review the changes anyways + the test code is reviewable independently of behavioral changes.

**Comment by rrazvan1** [comment]
> Can we merge this directly into #5204? I think that the tests should be in the same PR because they validate the changes introduced, and although it makes the resulting diff bigger I don't think it is actually more difficult to review because the tests are required to review the changes anyways + the test code is reviewable independently of behavioral changes.

Yes, they are related, but I thought the diff would've been too much. I prefer having less PR to maintain, so I will do it!
